examples/imagenet/train_imagenet_multi.py

```diff
@@ -68,6 +68,27 @@ def __call__(self, in_data):
         return img, label
 
 
+class FixedBatchDataset(chainer.dataset.DatasetMixin):
+    # Make the dataset size multiple of the batch-size by augmentation
+
+    def __init__(self, dataset, batchsize):
+        self.dataset = dataset
+        self.batchsize = batchsize
+        d = len(self.dataset)
+        self._len = ((d + batchsize - 1) // batchsize) * batchsize
+
+    def __len__(self):
+        return self._len
+
+    def get_example(self, idx):
+        if idx < len(self.dataset):
+            return self.dataset[idx]
+        else:
+            x_dummy, _ = self.dataset[0]
+            t_dummy = -1
```

Comment:
Ah, nice! Could you add a link to https://docs.chainer.org/en/stable/reference/generated/chainer.functions.softmax_cross_entropy.html to explain what -1 is? Also, it'd be better add `self.ignore_label`, e.g.,:

```
# `ignore_label` should be consistent with
# https://docs.chainer.org/en/stable/reference/generated/chainer.functions.softmax_cross_entropy.html
def __init__(self, dataset, batchsize, ignore_label=-1):
  self.ignore_label = ignore_label
```

Reply:
Thank you for comment. Modified the code.